pictures/tasks.py

```diff
@@ -33,11 +33,15 @@ def _process_picture(
     old = old or []
     storage = utils.reconstruct(*storage)
     if new:
-        with storage.open(file_name) as fs:
-            with Image.open(fs) as img:
-                for picture in new:
-                    picture = utils.reconstruct(*picture)
-                    picture.save(img)
+        try:
+            with storage.open(file_name) as fs:
+                with Image.open(fs) as img:
+                    for picture in new:
+                        picture = utils.reconstruct(*picture)
+                        picture.save(img)
+        except FileNotFoundError:
+            # The file no longer exists (for example, because it was deleted or replaced).
+            return
```

Comment:
Just silently ignoring missing files, will make it challenging to debug potential errors. I believe all message queues we use have a native behavior to fail without retrying on explicit exceptions. I'd recommend exploring that route.

Reply:
@codingjoe hm, this is a good suggestion!
I am struggling with figuring out the good way of doing so in django-rq (the interface is quite different from what I used to see in the other queues).
I added an example commit of how I see it with celery and dramatiq:
7c73c82f297750d7fa0989ec19aec29245780676

Reply:
@amureki my search only brought up this https://github.com/rq/rq/pull/1480/files